### DISCUSSION TITLE
macos 10.14 coapp damaged previous installation

### DISCUSSION THREAD

**Discussion by dieucode** [aclap-dev/video-downloadhelper#1245] [Help and Q&A]
### Your question

It said I had to have version 2 of the coapp to download that particular video, so I installed the one for macos14 hopping it would also work for macos 10.12 but the installation failed and now I have no coapp and I don't know which version to install. vdhcoapp-2.0.2-mac-x86_64-installer.pkg also failed

### Addon & CoApp version and details

Video DownloadHelper 9.1.0.44 (stable)
Video DownloadHelper
Version 9.1.0.44 (stable)
Langue du navigateur: fr
Généré pour la production
Généré le 2024-09-04
Options de génération: browser=chrome, linuxlic=false, noyt=true, target=google
⊖Plateforme
Plateforme Mac x86-64
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.5060.134 Safari/537.36 OPR/89.0.4447.91
⊖Appli compagnon non installée
Le test de l'appli compagnon a retourné: Specified native messaging host not found.. [Cliquer ici pour résoudre les problèmes](https://github.com/aclap-dev/video-downloadhelper/wiki/CoApp-not-recognized)

**Comment by dieucode** [comment]
Update : I installed coapp 1.6.3 successfully, however it is not detected (Specified native messaging host not found.) I restarted the browser and it's the same. I launched net.downloadhelper.coapp.app which is in the main application folder, and it did nothing.

**Comment by dieucode** [comment]
I need help

**Comment by dieucode** [comment]
I need help

**Comment by Mattmjs** [comment]
You need to go back to version 8.0.0.6 of the extension and disable updates. This can only be done on Firefox.
The other browsers don't keep older versions. Refer to this https://github.com/aclap-dev/video-downloadhelper/discussions/1102 where another user got it working.

**Reply by dieucode** [reply]
Thanks ! using the DMG instead of the pkg made it work. I don't understand why you talk about v8 and disabling updates.

The reason I had tried coapp v2 is because for most downloads it said it was required. 

Right now I have coapp v1.6.3 and I can't download much with it - what's the deal ?

**Reply by Mattmjs** [reply]
I say that because the newer version 9 extensions might tell you to install the latest version 2x companion app in order to
download something. But your old Mac can't do that because it's too outdated. This is the problem you might run into.
That's why I think you might need 8.0.0.6 for Firefox and turn off auto updates. This was the the last version that ran with v1.6.3.

**Reply by dieucode** [reply]
Oh so you mean that v8 would proceed to download instead of requesting coapp v2 ?

**Reply by Mattmjs** [reply]
Yes that's I mean, the only issue is do you have a license for Chrome. A Chrome license can't be used on Firefox.

**Reply by dieucode** [reply]
I never had a licence.

The old firefox versions are there https://addons.mozilla.org/en-US/firefox/addon/video-downloadhelper/versions/
But where are the chrome versions ?? I can't find a release forlder on github

**Reply by Mattmjs** [reply]
As I  already said the chrome and edge browsers don't keep older extensions on their 
respective stores. Only Mozilla does this for
Firefox. If there were old extension versions for
chrome and edge there is no switch to disable 
auto updates. Firefox has this toggle and it's 
one of the few things they do right.

**Reply by dieucode** [reply]
Yeah you said that, and I said that I am an Opera user.
We are on the official github, there is is a folder with the old coapp versions.
**But where are the old versions of VDH ??**

**Reply by Mattmjs** [reply]
None are kept on GitHub.

**Reply by GrampaWildWilly** [reply]
https://github.com/aclap-dev/video-downloadhelper/wiki/Older-versions

**Reply by dieucode** [reply]
Ok then I will make another post where I request someone to share an old version. 
Do you confirm that 8.0.0.6 is the most recent version that will download without coapp v2 ?

**Reply by Mattmjs** [reply]
That's the one that'll work without companion version 2x, I looked at the next version 8.1.4.0 and it says an update to
companion version 2.0.10 is required.

**Reply by GrampaWildWilly** [reply]
I would go to the page whose URL I just gave you.  Click the link there for "on our site for betas."  Then get the old beta 8.1.3.0a1.  I think the green button on that page will get you the latest VDH version 9.  I'm not sure.  You could try it.  It might be a VDH 8 beta newer than 8.1.3.0a1.  I'm pretty sure there were betas of VDH 8 after 8.1.3.0a1 but Paul stopped bothering to keep that archive updated, despite my complaints about it.  But you're best to get the very latest beta you can.  There were bugs in the general availability versions available through the Mozilla web site that were corrected in the various betas.  I think we got as high as 8.2.x.x or maybe even 8.3.x.x in the betas before VDH 9 came out.  I long ago discarded my .xpi file for the last VDH 8 beta so I can't give it to you.  Maybe some other person who is even more of a pack rat than I am can do that.

There is a version 8.2.2.8 on Mozilla.  As I recall, there was a period during which the VDH 8 versions had some new features that were available only if you ran a CoApp 2.x.  But you could still use CoApp 1.6.3 with it anyway.  You just couldn't get the improvements provided by CoApp 2 but it would still work with CoApp 1.6.3.  You should experiment with the various combinations of VDH 8 & CoApp 1.6.3.  The worst that will happen is you'll get an error message.  Start with VDH 8.2.2.8 & if that doesn't like CoApp 1.6.3, try the next older one.  Keep going until you find one that likes CoApp 1.6.3.  I'm not going to certify that 8.0.0.6 is the latest you can get away with.  I'm just another user.  You have to experiment & find out for yourself.  If the last general availability beta that works is a VDH 8.1.x, then I would get the beta as I described in my first paragraph of this post.  You might have to start with 8.1.3.0a1 beta & work your way backwards from there.  Whatever you settle on, a beta is preferable to a general availability version because the betas contain bug corrections that are missing from the general availability versions.  Give yourself the best chance of getting successful downloads.  And as Matt has pointed out more than once, turn off automatic updates in the Firefox settings for whatever VDH you finally settle on.

Once you find a combination you think will work, be prepared for downloads to still fail anyway.  There's been a lot of progress in the product since VDH 8.  A lot of failing web sites now work.  But if you're on an older version, you have to be prepared to just accept some failures.  You won't get any help here, either.  People here are using the latest versions & don't have the older versions to even experiment with.  The best anybody here will be able to do is tell you that with VDH 9.x & CoApp 2.x, it works fine.  In that case, you'll just have to accept that you can't download whatever it is.  That's life.

**Reply by dieucode** [reply]
Those betas are firefox only.
I need opera/chrome, and I found it : https://www.crx4chrome.com/history/51037/
This link should be pinned.

**Reply by GrampaWildWilly** [reply]
My bad.  Sorry.  Glad you found what you need.

**Reply by dieucode** [reply]
Thanks both for your help

**Reply by Mattmjs** [reply]
As I already said chrome has no toggle to disable auto updates so whatever you
add will eventually update on it's own. Opera I don't know if they have a toggle to turn off auto updates.
In any event good luck with it.

**Reply by dieucode** [reply]
Damn I spoke too fast.

So it let me download with the browser however the video had no sound.
I tried again and chose download with coapp, but while it started it didn't finish and upgraded VDH to the last version ! 
I can't find a setting to block updates. Can I tweak the extension files to change that ?

**Reply by Mattmjs** [reply]
I thought I was done with this discussion, there is a manual hack to edit the update url
https://stackoverflow.com/questions/27657617/how-to-disable-google-chrome-extension-autoupdate

Video with no sound : https://groups.google.com/g/video-downloadhelper-q-and-a/c/nGktdtm2HZ8/m/XLIsfBjXCAAJ

**Comment by Mattmjs** [comment]
Here is another discussion to refer to https://github.com/aclap-dev/video-downloadhelper/discussions/288
